### PR TITLE
lang: add additional `require_x` comparison macros

### LINKED ISSUE #1591: lang: add more require macros
- require_neq
- require_keys_neq
- require_gt
- require_gte

Add a default error code for each of these in case no custom error is provided.

blocked by #1586

### PR DESCRIPTION
**Pull request by callensm**
resolves #1591

### CODE REVIEW

File: lang/src/lib.rs [hunk 1]
```diff
@@ -427,6 +457,94 @@ macro_rules! require_keys_eq {
     };
 }
 
+/// Ensures two pubkeys values are not equal.
```
Comment (paul-schaaf) [inline lang/src/lib.rs:460]: `Ensures two pubkeys are not equal.`

**Comment by paul-schaaf** [comment]
im assuming this is ready for review? If so, could u add tests for each

**Comment by callensm** [comment]
> im assuming this is ready for review? If so, could u add tests for each

done!

**Review by paul-schaaf — APPROVED** [review]

**Comment by paul-schaaf** [comment]
thank you!

**Pull request merged**